src/test/java/org/apache/drillui/test/framework/pages/BasePage.java

```diff
@@ -69,4 +79,101 @@ protected static void sendText(WebElement element, String text) {
     element.sendKeys(Keys.CONTROL, "v");
   }
 
+  protected void waitForCondition(Function<WebDriver, Boolean> condition) {
+    waitForCondition(condition, TestProperties.getInt("DEFAULT_TIMEOUT"));
+  }
+
+  protected void waitForCondition(Function<WebDriver, Boolean> condition, int timeOut) {
+    if (!condition.apply(getDriver())) {
+      new WebDriverWait(getDriver(), timeOut)
+          .until(condition);
+    }
+  }
+
+  protected void waitForCondition(Function<WebDriver, Boolean> condition, Runnable workaroundAction) {
+    int timeStep = 1;
+    int elapsed = 0;
+    boolean success = false;
+    while (!condition.apply(getDriver()) && elapsed < TestProperties.getInt("DEFAULT_TIMEOUT") - 1) {
```

Comment:
You should use Guava Stopwatch class here to handle elapsed time.

For example:
<dependency>
            <groupId>com.google.guava</groupId>
            <artifactId>guava</artifactId>
            <version>23.0</version>
</dependency>

import com.google.common.base.Stopwatch;
import java.util.concurrent.TimeUnit;
public class Main {
  public static void main(String[] args) {
    final Stopwatch stopwatch = Stopwatch.createStarted();
    System.out.println("Started at: " + stopwatch.elapsed(TimeUnit.MILLISECONDS) + " msec");
    try { Thread.sleep(2000); } catch (InterruptedException e) {e.printStackTrace();}
    System.out.println("Elapsed time: " + stopwatch.elapsed(TimeUnit.MILLISECONDS) + " msec");
  }
}

```suggestion
    final Stopwatch stopwatch = Stopwatch.createStarted();
    while (!condition.apply(getDriver()) && stopwatch.elapsed(TimeUnit.SECONDS) < TestProperties.getInt("DEFAULT_TIMEOUT") - 1) {
```

Reply:
Done.